lib/utils/deploy-legacy.coffee

```diff
@@ -0,0 +1,145 @@
+Promise = require('bluebird')
+
+getBuilderPushEndpoint = (baseUrl, owner, app) ->
+	querystring = require('querystring')
+	args = querystring.stringify({ owner, app })
+	"https://builder.#{baseUrl}/v1/push?#{args}"
+
+getBuilderLogPushEndpoint = (baseUrl, buildId, owner, app) ->
+	querystring = require('querystring')
+	args = querystring.stringify({ owner, app, buildId })
+	"https://builder.#{baseUrl}/v1/pushLogs?#{args}"
+
+bufferImage = (docker, imageId, bufferFile) ->
+	Promise = require('bluebird')
+	streamUtils = require('./streams')
+
+	image = docker.getImage(imageId)
+	imageMetadata = image.inspect()
+
+	Promise.join image.get(), imageMetadata.get('Size'), (imageStream, imageSize) ->
+		streamUtils.buffer(imageStream, bufferFile)
+		.tap (bufferedStream) ->
+			bufferedStream.length = imageSize
+
+showPushProgress = (message) ->
+	visuals = require('resin-cli-visuals')
+	progressBar = new visuals.Progress(message)
+	progressBar.update({ percentage: 0 })
+	return progressBar
+
+forwardStatusMessageFromRemote = (logger, msg) ->
+	msg.split(/\r\n|\n/).forEach (line) ->
+		if /^Warning: This endpoint is deprecated/.test(line)
+			return
```

Comment:
I don't love this... If we're effectively undeprecating the legacy endpoint, shouldn't we just undeprecate it in the builder itself?

It's going to be ridiculous & annoying when really really want to deprecate it in future because it's actually going to be 100% removed, and we have to make sure our message doesn't include the above or some CLIs won't show it.